Add DeckyFileServer

grimkor opened the pull request:
# DeckyFileServer

A decky plugin that allows you to quickly browse to files on your steam deck from a browser on another device.

## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.


### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.


## Testing

- [ ] Tested on SteamOS Stable/Beta Update Channel.

PartyWumpus commented:
When testing, it says the certificate is not valid (which is probably correct). I assume this isn't really fixable, maybe put a warning somewhere in the UI. Also, the address `https://steamdeck:port/` also works and would be easier for people to type, although I haven't tested that on a windows device. (Also the website could do with a home button if possible, but that's more of a future thing I'd like)

In the settings page there's a typo in the title. Also, the select folder button has a line underneath, iirc this is fixed by the button's `bottomSeparator` value to 'none' (from https://github.com/SteamDeckHomebrew/decky-frontend-lib/blob/5ffa14bec8ec3a4d192be812175debd708486839/src/deck-components/Item.tsx#LL8C3-L8C18). If you plan to add many more settings it might be worth making it a full-screen page, but is fine as is.
![image](https://github.com/SteamDeckHomebrew/decky-plugin-database/assets/48649272/2c4edb63-a051-4ac1-9a0f-d3cdcc95051b)

Other than those tiny UX issues, it works flawlessly! Great plugin, will be very useful. (also the file-picker doesn't work on stable decky but that's decky's fault so don't worry about that)

PartyWumpus commented:
Also the port number errors below 1023, but should also probably error *above* 65535

PartyWumpus reviewed (approved):
None of the above changes are required though, so getting my approval. (codebase looks good)

grimkor commented:
> 

Good point on all of these, I've made the changes and I'll test tonight. I went for a dialog window for the settings at the moment as the settings looked a bit lost in the whole page. I'll think about how to prepare users for the known certificate warning, I'll find a way to weave it into the plugin without pestering people with lots of alerts.

I had a plan to add a "Only use this on known networks", I could add it to that as a first-time user notification.

Appreciate the time to look through this!

grimkor commented:
* Added port range restriction for both high and low value.
* Fixed typo.
* Cleaned up a little while I was there.
* Added initial warning message that must be accepted before enabling for the first time.
![20230623185555_1](https://github.com/SteamDeckHomebrew/decky-plugin-database/assets/19330133/ab01dd02-095d-4056-8687-e851d0332a77)

PartyWumpus commented:
your deployment "failed", but it also uploaded to the store completely fine. All changes seem to work and code changes seem reasonable. LGTM! (I did also notice that it defaults to `/home/deck/` which may not be correct, but that is a very low priority fix as decky is only really designed for the steam deck)

grimkor commented:
> your deployment "failed", but it also uploaded to the store completely fine. All changes seem to work and code changes seem reasonable. LGTM! (I did also notice that it defaults to `/home/deck/` which may not be correct, but that is a very low priority fix as decky is only really designed for the steam deck)

Deployment failed successfully!

Is `/home/deck` not the default for all decks? If there's a better option I can do it in a later update

RodoMa92 commented:
> > your deployment "failed", but it also uploaded to the store completely fine. All changes seem to work and code changes seem reasonable. LGTM! (I did also notice that it defaults to `/home/deck/` which may not be correct, but that is a very low priority fix as decky is only really designed for the steam deck)
> 
> Deployment failed successfully!
> 
> Is `/home/deck` not the default for all decks? If there's a better option I can do it in a later update

Yes for all decks, but since it seems that a lot of people aren't running it on deck, the home directory might be different.

There is an initial implementation for resolving in decky loader itself, but I do not know if it's accessible from plugins.

grimkor commented:
I'll see if I can make a better default next update I make, maybe there's a
decky plugin variable I can use.

Appreciate working with this and the feedback!

On Sun, 25 Jun 2023, 18:08 Marco Rodolfi, ***@***.***> wrote:

> your deployment "failed", but it also uploaded to the store completely
> fine. All changes seem to work and code changes seem reasonable. LGTM! (I
> did also notice that it defaults to /home/deck/ which may not be correct,
> but that is a very low priority fix as decky is only really designed for
> the steam deck)
>
> Deployment failed successfully!
>
> Is /home/deck not the default for all decks? If there's a better option I
> can do it in a later update
>
> Yes for all decks, but since it seems that a lot of people aren't running
> it on deck, the home directory might be different.
>
> There is an initial implementation for resolving in decky loader itself,
> but I do not know if it's accessible from plugins.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/387#issuecomment-1606165309>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AETPIVNPJPNOR3X4TRURTDLXNBV7RANCNFSM6AAAAAAZKZB6HY>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
